plugins/woocommerce/client/blocks/assets/js/types/type-defs/cart-response.ts

```diff
@@ -163,6 +163,11 @@ export interface CartResponseExtensionItem {
 	[ key: string ]: unknown;
 }
 
+export interface ShippingAddressFieldsForShippingRates
+	extends Omit< ResponseBaseAddress, 'address_1' | 'address_2' > {
+	[ key: string ]: string;
+}
+
```

Comment:
I don't think this type is correct because it's saying that `ShippingAddressFieldsForShippingRates` is an object of strings, keyed by strings, whereas it is an array of strings. We should probably use the keys of `ResponseBaseAddress`without `address_1` and `address_2`.

e.g. 
```suggestion
export type ShippingAddressFieldsForShippingRates = keyof Omit<
	ResponseBaseAddress,
	'address_1' | 'address_2'
>;
```

Reply:
Nice catch! I've actually based it on BaseAddress, as I've learned that the response types are for data coming from Store API. See https://github.com/woocommerce/woocommerce/pull/58030/commits/f8200fec0b381e135c50465ea3a4b78a38125975